### PR TITLE
Adds support for Intel FPGA Nios V/m

### PR DESCRIPTION
**Pull request by sweeaun**
Adds support for Intel FPGA Nios V/m 

This PR adds support for Intel FPGA Nios V/m SoC and board, board system design is based on Intel FPGA Design Store Nios® V/m Hello World Example Design system.

More information about Intel FPGA Nios V/m and Hello World Example Design system can be obtained here:
https://www.intel.com/content/www/us/en/products/details/fpga/nios-processor/v.html
https://www.intel.com/content/www/us/en/design-example/763960/arria10-niosv-based-helloworld-example-design-on-arria10-devkit.html

For testing, it was verified with following samples
hello_world
synchronization
philosophers

### CODE REVIEW

File: dts/riscv/niosv/niosv-m.dtsi [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+		mtimer: machine-timer@90000 {
+			compatible = "niosv-machine-timer";
+			reg = <0x90000 0x10>;
+			interrupts = <7>;
+		};
+
+		uart0: serial@90078 {
+			compatible = "altr,jtag-uart";
+			status = "disable";
```
Comment (fkokosinski) [inline dts/riscv/niosv/niosv-m.dtsi:53]: It looks like at least some of the build fails are related to this node. You would need either add interrupts here, or disable `CONFIG_UART_INTERRUPT_DRIVEN`.

Comment (sweeaun) [reply dts/riscv/niosv/niosv-m.dtsi:54]: Got it. I just amended dtsi file to add corresponding interrupts number. Thanks.

**Review by teburd — DISMISSED** [review]

**Review by fkokosinski — DISMISSED** [review]
LGTM

File: boards/riscv/niosv_m/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	JTAG UART Intel® FPGA IP
+	On-Chip Memory Intel® FPGA IP
+
+Nios® V/m hello world example design system
+===========================================
+
+Prebuilt Nios® V/m hello world example design system is available in Intel FPGA Design store.
+- https://www.intel.com/content/www/us/en/support/programmable/support-resources/design-examples/design-store.html?s=Newest
+
+.. code-block:: console
```
Comment (edersondisouza) [inline boards/riscv/niosv_m/doc/index.rst:23]: Why these empty code blocks? Maybe you want to check how the generated docs look like.

**Review by edersondisouza — COMMENTED** [review]
Minor comment about docs.

File: boards/riscv/niosv_m/doc/index.rst (hunk 2 above)
Comment (sweeaun) [reply boards/riscv/niosv_m/doc/index.rst]: The empty code blocks have been removed. Thanks.

**Review by teburd — DISMISSED** [review]

**Review by edersondisouza — DISMISSED** [review]

**Comment by teburd** [comment]
You don’t need to rebase unless there are conflicts, it won’t go in until 3.3 is released, will re approve once that happens. PR looks good! Looking forward to trying it out.

**Review by fkokosinski — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**